src/main/webapp/create-cache/create-cache.controller.js

```diff
@@ -0,0 +1,196 @@
+'use strict';
+
+angular.module('managementConsole')
+  .controller('createCacheCtrl', [
+    '$scope',
+    '$state',
+    '$stateParams',
+    'utils',
+    'modelController',
+    'cacheCreateController',
+    function ($scope, $state, $stateParams, utils, modelController, cacheCreateController) {
+      //-- Variables --//
+
+      if (!modelController.isAuthenticated()) {
+        $state.go('/logout');
+      }
+      if (!$stateParams.clusterName) {
+        $state.go('error404');
+      }
+
+      var server = modelController.getServer();
+      var clusters = server.getClusters();
+      $scope.currentCluster = server.getCluster(clusters, $stateParams.clusterName);
+      $scope.newlyCreatedCache = {};
+      $scope.selectedCacheType = 'distributed-cache';
+      $scope.isCreateEnabled = false;
+
+      $scope.createReplicatedOrDistributedCache = function (cache, address, cacheType) {
+        var promise = null;
+        if (cacheType === 'distributed-cache') {
+          promise = cacheCreateController.createDistributedCache(address, cache.data.mode, cache.data.module, cache.data['remote-cache'],
+            cache.data.segments, cache.data.indexing, cache.data['auto-config'], cache.data.statistics, cache.data['remote-timeout'],
+            cache.data['capacity-factor'], cache.data.batching, cache.data.start, cache.data['l1-lifespan'], cache.data['remote-site'],
+            cache.data['jndi-name'], cache.data['queue-size'], cache.data['queue-flush-interval'], cache.data.owners);
+        } else if (cacheType === 'replicated-cache') {
+          promise = cacheCreateController.createReplicatedCache(address, cache.data.mode, cache.data.module, cache.data['remote-cache'],
+            cache.data.indexing, cache.data['auto-config'], cache.data.statistics, cache.data['remote-timeout'],
+            cache.data.batching, cache.data.start, cache.data['remote-site'],
+            cache.data['jndi-name'], cache.data['queue-size'], cache.data['queue-flush-interval']);
+        }
+        promise.then(function () {
+          var lockingAddress = angular.copy(address);
```

Comment:
There seems to some amount of repetition in this code, maybe it would be good to generalize it (with some function)? Or would generalazation here would be overengineering and reduce flexibility? Just giving an idea!